backend/danswer/document_index/vespa/index.py

```diff
@@ -377,6 +377,93 @@ def update(self, update_requests: list[UpdateRequest]) -> None:
             time.monotonic() - update_start,
         )
 
+    def update_single(self, update_request: UpdateRequest) -> None:
```

Comment:
this is a decent amount of duplication with the standard `update`. Why do we need this?

Reply:
The single doc version is a great deal simpler to step through and debug, and I believe most of what the update method is doing with various threadpools, etc ... we should be able to achieve similar parallelism with appropriate usage of a single httpx connection pool dedicated to vespa and parallelism through celery.  Keeping regular update around for now, but it could be deprecated once this gets enough battle testing.